dot/parachain/candidate-validation/candidate_validation.go

```diff
@@ -80,7 +80,18 @@ func (cv *CandidateValidation) processMessages(wg *sync.WaitGroup) {
 			logger.Debugf("received message %v", msg)
 			switch msg := msg.(type) {
 			case ValidateFromChainState:
-				// TODO: implement functionality to handle ValidateFromChainState, see issue #3919
+				result, err := validateFromChainState(cv.RuntimeInstance, msg.Pov, msg.CandidateReceipt)
+				if err != nil {
+					logger.Errorf("failed to validate from chain state: %w", err)
+					msg.Ch <- parachaintypes.OverseerFuncRes[ValidationResult]{
+						Data: *result,
+						Err:  err,
+					}
```

Comment:
If we are not able to validate the candidate, mostly result would be nil.
dereferencing nil would cause panic here.

also, there is an error so we don't need to set Data field.
```suggestion
					msg.Ch <- parachaintypes.OverseerFuncRes[ValidationResult]{
						Err:  err,
					}
```

Reply:
Good point, updated.